### PR TITLE
feature/delete-movie

### PR DESCRIPTION
**Pull request by andjela11**

### CODE REVIEW

File: Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    private readonly IDataContext _context;
+
+    public DeleteMovieCommandHandler(IDataContext context)
+    {
+        this._context = context;
+    }
+
+    public async Task Handle(DeleteMovieCommand request, CancellationToken cancellationToken)
+    {
+        var movie = await _context.Movies.FindAsync(request.Id);
```
Comment (ares-5) [inline Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs]: What is the appropriate way to search for a movie. What if we have more than 1 movie with same id

File: Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        if (movie is null)
+        {
+            throw new EntityNotFoundException("Movie not found");
```
Comment (ares-5) [inline Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs]: This message can be improved. Use string interpolation to also pass id which was passed

File: Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        }
+
+        this._context.Movies.Remove(movie);
+        await this._context.SaveChangesAsync(cancellationToken);
```
Comment (ares-5) [inline Application/Features/Commands/DeleteMovie/DeleteMovieCommandHandler.cs:28]: I think entity framework is returning successful/unsuccessful operation for SaveChangesAsync, maybe you can use that flag to see if operation was successful

File: Application/Features/Commands/DeleteMovie/DeleteMovieCommandValidation.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using FluentValidation;
+
+namespace Application.Features.Commands.DeleteMovie;
+
+public class DeleteMovieCommandValidation : AbstractValidator<DeleteMovieCommand>
+{
+    public DeleteMovieCommandValidation()
+    {
+        RuleFor(x => x.Id).NotNull();
```
Comment (ares-5) [inline Application/Features/Commands/DeleteMovie/DeleteMovieCommandValidation.cs]: What if I pass negative number, or zero?

File: Application/Features/Commands/DeleteMovie/DeleteMovieCommandValidation.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using FluentValidation;
+
+namespace Application.Features.Commands.DeleteMovie;
+
+public class DeleteMovieCommandValidation : AbstractValidator<DeleteMovieCommand>
```
Comment (ares-5) [inline Application/Features/Commands/DeleteMovie/DeleteMovieCommandValidation.cs]: Class suffix shouldn't be Validation. It should be Validator

File: WebAPI/Controllers/MoviesController.cs [hunk 6]
```diff
@@ -82,4 +83,22 @@ public async Task<ActionResult> CreateMovieAsync([FromBody] CreateMovieDto movie
...
+    /// </summary>
+    /// <param name="id">Movie id</param>
+    /// <returns></returns>
+    [HttpDelete("{id:int}")]
+    [SwaggerResponse((int)HttpStatusCode.NoContent)]
+    [SwaggerResponse((int)HttpStatusCode.UnprocessableEntity, Description = "Unsuccessful validation")]
+    [SwaggerResponse((int)HttpStatusCode.InternalServerError, Description = "Unexpected event occurred")]
+    public async Task<ActionResult> DeleteMovieAsync(int id)
+    {
+        var req = new DeleteMovieCommand(id);
```
Comment (ares-5) [inline WebAPI/Controllers/MoviesController.cs]: req? Is that appropriate name?

File: WebAPI/Middleware/ExceptionMiddleware.cs [hunk 7]
```diff
@@ -44,33 +45,37 @@ public async Task InvokeAsync(HttpContext context)
...
+            await GenerateExceptionResponse(context, (int)HttpStatusCode.InternalServerError, e, "Something went wrong");
+        }
+    }
 
-            var response = this._env.IsDevelopment()
-                ? new ErrorDetails(context.Response.StatusCode, e.Message, e.StackTrace)
-                : new ErrorDetails(context.Response.StatusCode, e.Message, "Something went wrong");
-            
-            var opt = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+    private async Task GenerateExceptionResponse(HttpContext context, int statusCode, Exception e, string details, [Optional] Dictionary<string, string[]> validationErrors)
```
Comment (ares-5) [inline WebAPI/Middleware/ExceptionMiddleware.cs]: Good job having logic in one place, try ordering the parameters a bit better, and split them by line. It's hard to read if all of them are in the same line.

File: WebAPI/Middleware/ExceptionMiddleware.cs [hunk 8]
```diff
@@ -44,33 +45,37 @@ public async Task InvokeAsync(HttpContext context)
...
+        if (validationErrors is not null)
+        {
+            response.ValidationErrors = validationErrors;
         }
+
+        var opt = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+
+        var json = JsonSerializer.Serialize(response, opt);
+
+        await context.Response.WriteAsync(json);
```
Comment (ares-5) [inline WebAPI/Middleware/ExceptionMiddleware.cs:84]: Did you test it out? Still works like before?

**Review by ares-5 — CHANGES_REQUESTED** [review]

File: Application/Features/Commands/DeleteMovie/DeleteMovieCommandValidation.cs (hunk 4 above)
Comment (andjela11) [reply Application/Features/Commands/DeleteMovie/DeleteMovieCommandValidation.cs]: 404 not found exception will be thrown

File: WebAPI/Middleware/ExceptionMiddleware.cs (hunk 8 above)
Comment (andjela11) [reply WebAPI/Middleware/ExceptionMiddleware.cs:84]: I tested it, it works.

**Review by ares-5 — APPROVED** [review]

**Pull request merged**